src/packs/nzism-v36-1022-20.ts

```diff
@@ -0,0 +1,945 @@
+import { KMSBackingKeyRotationEnabled } from '../rules/kms';
+import { LambdaFunctionPublicAccessProhibited } from '../rules/lambda';
+import {
+  OpenSearchEncryptedAtRest,
+  OpenSearchInVPCOnly,
+  OpenSearchNodeToNodeEncryption,
+} from '../rules/opensearch';
+import {
+  RDSAutomaticMinorVersionUpgradeEnabled,
+  RDSInBackupPlan,
+  RDSInstanceDeletionProtectionEnabled,
+  RDSInstancePublicAccess,
+  RDSLoggingEnabled,
+  RDSMultiAZSupport,
+  RDSStorageEncrypted,
+} from '../rules/rds';
+import {
+  RedshiftRequireTlsSSL,
+  RedshiftBackupEnabled,
+  RedshiftClusterConfiguration,
+  RedshiftClusterMaintenanceSettings,
+  RedshiftClusterPublicAccess,
+} from '../rules/redshift';
+import {
+  S3BucketLevelPublicAccessProhibited,
+  S3BucketLoggingEnabled,
+  S3BucketPublicReadProhibited,
+  S3BucketPublicWriteProhibited,
+  S3BucketServerSideEncryptionEnabled,
+  S3BucketSSLRequestsOnly,
+  S3BucketVersioningEnabled,
+  S3DefaultEncryptionKMS,
+} from '../rules/s3';
+import {
+  SageMakerEndpointConfigurationKMSKeyConfigured,
+  SageMakerNotebookInstanceKMSKeyConfigured,
+  SageMakerNotebookNoDirectInternetAccess,
+} from '../rules/sagemaker';
+import { SecretsManagerUsingKMSKey } from '../rules/secretsmanager';
+import { SNSEncryptedKMS } from '../rules/sns';
+import {
+  VPCFlowLogsEnabled,
+  VPCDefaultSecurityGroupClosed,
+} from '../rules/vpc';
+import { WAFv2LoggingEnabled } from '../rules/waf';
+
+/**
+ * Check for NZISM v36-1022-20 compliance.
+ * Based on the NZISM v36-1022-20: <URL To be updated once published>
+ * */
+
+export class NZISM36Checks extends NagPack {
+  constructor(props?: NagPackProps) {
+    super(props);
+    this.packName = 'NZISM3.6';
+  }
+  public visit(node: IConstruct): void {
+    if (node instanceof CfnResource) {
+      this.checkELB(node);
+      this.checkAPIGW(node);
+      this.checkCloudTrail(node);
+      this.checkCloudFront(node);
+      this.checkKMS(node);
+      this.checkCloudWatch(node);
+      this.checkRDS(node);
+      this.checkECS(node);
+      this.checkEFS(node);
+      this.checkDMS(node);
+      this.checkDynamoDB(node);
+      this.checkEC2(node);
+      this.checkElasticBeanstalk(node);
+      this.checkElastiCache(node);
+      this.checkLambda(node);
+      this.checkOpenSearch(node);
+      this.checkIAM(node);
+      this.checkRedshift(node);
+      this.checkS3(node);
+      this.checkSageMaker(node);
+      this.checkSecretsManager(node);
+      this.checkSNS(node);
+      this.checkVPC(node);
+      this.checkWAF(node);
+    }
+  }
+
+  /**
+   * Check API Gateway Resources
+   * E
+   * @param node the CfnResource to check
+   * @param ignores list of ignores for the resource
+   */
+  private checkAPIGW(node: CfnResource): void {
+    this.applyRule({
+      ruleSuffixOverride: 'API_GATEWAY_LOGGING',
+      info: 'The API Gateway stage does not have execution logging enabled for all methods. | SHOULD 16.6.10.C.02[CID:2013], MUST 23.5.11.C.01[CID:7496] ',
```

Comment:
Continuing from this comment  https://github.com/cdklabs/cdk-nag/pull/1068#discussion_r1010839379 in the previous PR.

`cdk-nag` just uses the control id (like [here](https://docs.aws.amazon.com/config/latest/developerguide/operational-best-practices-for-nzism.html)).

I read through the NZISM [ISM](https://www.nzism.gcsb.govt.nz/ism-document) document to verify that all  the Control Ids are unique, and that the reference material can be located from just the control id

Reply:
Yes, they are Unique, but I've never actually heard anyone ( i spend my world in NZ Government IT ), refer to the CID's, they always will give you the chapter markings and thats what they refer to..     While you can search for them on CID, and get them that way..  Given this is an info line, can't we have both?